pkgs/shells/marcel/default.nix

```
@@ -0,0 +1,56 @@
+{ lib
+, stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, setuptools
+, dill
+, psutil
+, makeWrapper
+, python3
+, bash
+}:
+
+buildPythonPackage rec {
+  pname = "marcel";
+  version = "0.17.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "geophile";
+    repo = "marcel";
+    rev = "v${version}";
+    hash = "sha256-JMDZiZcsSXWxyx2ZJMihlzl62FPk+V8HWejQwZObpUs=";
+  };
+
+  propagatedBuildInputs = [
+    dill
+    psutil
+    bash
+  ];
+
+  doCheck = false; # cause the tests provided by marcel will read/write $HOME/.local/share/marcel
+
+  prePatch = ''
+    substituteInPlace marcel/op/bash.py experiments/signal_propagation.py marcel/op/edit.py experiments/cp_behavior.py experiments/popen_stdout_stderr.py \
+        --replace '/bin/bash' ${bash}/bin/bash
+  '';
+
+  postInstall = ''
+    cat > $out/bin/marcel <<EOF
+    #!${bash}/bin/bash
+    exec python3 -m marcel.main "\$@"
+    EOF
+    chmod +x $out/bin/marcel
+    wrapProgram $out/bin/marcel \
+    --prefix PYTHONPATH : $out/${python3.sitePackages}
+  '';
```

```suggestion
  postFixup = ''
    substituteInPlace $out/bin/marcel \
        --replace python3 ${python3}/bin/python3
    wrapProgram $out/bin/marcel \
        --prefix PYTHONPATH : "$program_PYTHONPATH"
  '';
```
There's no need to manually recreate the `marcel` script, it is automatically done.

Just having `marcel`'s own `PYTHONPATH` is not enough, because there are dependencies.
`$program_PYTHONPATH` is a hacky way around the fact that this is not actually a python script.
`$program_PYTHONPATH` is set after `python` wraps all python scripts. This is why I put this into `postFixup` because that runs after that.

---

It works! Thank you!